src/uwtools/config/tools.py

```diff
@@ -110,6 +110,29 @@ def realize_config(
     return input_obj.data
 
 
+def walk_key_path(config: dict, key_path: list[str]) -> tuple[dict, str]:
+    """
+    Navigate to the sub-config at the end of the path of given keys.
+
+    :param config: A config.
+    :param key_path: Path of keys to subsection of config file.
+    :return: The sub-config and a string representation of the key path.
+    """
+    keys = []
+    pathstr = "<unknown>"
+    for key in key_path:
+        keys.append(key)
+        pathstr = " -> ".join(keys)
+        try:
+            subconfig = config[key]
+        except KeyError as e:
+            raise log_and_error(f"Bad config path: {pathstr}") from e
+        if not isinstance(subconfig, dict):
+            raise log_and_error(f"Value at {pathstr} must be a dictionary")
+        config = subconfig
+    return config, pathstr
```

Comment:
This function was factored out of `_print_config_section()`, below. It has its own, new unit tests, and is also called from the `Assets` class constructor to better deal with its key-path-walking needs.